### PR TITLE
fixed issue check for logged_in status on /settings #5923

### LINKED ISSUE #5923: Check for logged_in status on /settings
It looks like we aren't checking if people are logged in on https://publiclab.org/settings, causing an error when you load the page without logging in. We should catch and redirect; the controller is here:

I don't actually see a method for settings in https://github.com/publiclab/plots2/blob/master/app/controllers/users_controller.rb -- but routes.rb does say there is one. Maybe it's just an HTML file with no controller action? Probably. 

https://github.com/publiclab/plots2/blob/master/app/views/users/settings.html.erb

So maybe a change on this line would just enable a login check even though we don't have an explicit controller action:

https://github.com/publiclab/plots2/blob/master/app/controllers/users_controller.rb#L3

If not, we should just add this to make it explicit:

```ruby
def settings
end
```

### PR DESCRIPTION
**Pull request by yprasant0**
Fixes #5923  (<=== Add issue number here)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [ ] PR is descriptively titled 📑 and links the original issue above 🔗
* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [ ] code is in uniquely-named feature branch and has no merge conflicts 📁
* [ ] screenshots/GIFs are attached 📎 in case of UI updation
* [ ] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by yprasant0** [comment]
@jywarren @SidharthBansal  @ebarry can you help me this pull request I don't know why Travis check fails.

**Comment by SidharthBansal** [comment]
Restarted the build. Let's see!

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -399,7 +399,7 @@ def save_settings
 
     flash[:notice] = "Settings updated successfully!"
     render js: "window.location.reload()"
-  end
+    end
```
Comment (SidharthBansal) [inline app/controllers/users_controller.rb]: Not required

**Comment by SidharthBansal** [comment]
Please remove ``test sqlite journal`` file.

**Comment by yprasant0** [comment]
@SidharthBansal can you have a look, Travis still fails. I deleted the test.Sqlite file

**Comment by jywarren** [comment]
Hi @yprasant0 -- this just needs to be rebased to incorporate the just-fixed timing issue. Here are some resources on rebasing, do you think you can do that? 

You may be able to skip step 2 if your master is still in sync with publiclab's master: https://publiclab.org/wiki/contributing-to-public-lab-software#Rewinding+the+master+branch

Thanks!

**Comment by IshaGupta18** [comment]
@yprasant0 I have restarted Travis, lets see if it passes now!

**Comment by jywarren** [comment]
You can also just run `git pull https://github.com/publiclab/plots2.git master` and merge the changes, i think!

**Comment by yprasant0** [comment]
Phew!! It passes. Thank you @IshaGupta18 and @jywarren

**Comment by cesswairimu** [comment]
Hi @yprasant0, I just tested this locally and its working great :+1: .. Thanks for working on this. I see you have touched some files/lines that should should not be affected on this change. Do you think maybe you could check them out from the commit and just have the crucial changes on this? Let me know if I can help in any way. Thanks again.

**Comment by SidharthBansal** [comment]
As the person is inactive for more than a month, I am closing the PR. In case you want to push changes please feel free to open a new PR OR reopen this PR and add additional changes to it.
Thanks for contributing on Public Lab

**Pull request closed without merging**